lib/ClangImporter/ClangImporter.cpp

```diff
@@ -430,49 +430,61 @@ static void
 getNormalInvocationArguments(std::vector<std::string> &invocationArgStrs,
                              ASTContext &ctx,
                              const ClangImporterOptions &importerOpts) {
-  const llvm::Triple &triple = ctx.LangOpts.Target;
+  const auto &LangOpts = ctx.LangOpts;
+  const llvm::Triple &triple = LangOpts.Target;
   SearchPathOptions &searchPathOpts = ctx.SearchPathOpts;
 
   auto languageVersion = ctx.LangOpts.EffectiveLanguageVersion;
 
-  if (llvm::sys::path::extension(importerOpts.BridgingHeader).endswith(
-        PCH_EXTENSION)) {
-    invocationArgStrs.insert(
-      invocationArgStrs.end(),
-        { "-include-pch", importerOpts.BridgingHeader }
-    );
+  if (llvm::sys::path::extension(importerOpts.BridgingHeader)
+          .endswith(PCH_EXTENSION)) {
+    invocationArgStrs.insert(invocationArgStrs.end(), {
+        "-include-pch", importerOpts.BridgingHeader
+    });
   }
 
   // Construct the invocation arguments for the current target.
   // Add target-independent options first.
-  invocationArgStrs.insert(
-      invocationArgStrs.end(),
-      {
+  invocationArgStrs.insert(invocationArgStrs.end(), {
+      // Enable modules
+      "-fmodules",
+      "-Werror=non-modular-include-in-framework-module",
+      "-Xclang", "-fmodule-feature", "-Xclang", "swift",
 
-          // Enable modules
-          "-fmodules",
-          "-Werror=non-modular-include-in-framework-module",
-          "-Xclang", "-fmodule-feature", "-Xclang", "swift",
+      // Don't emit LLVM IR.
+      "-fsyntax-only",
 
-          // Don't emit LLVM IR.
-          "-fsyntax-only",
+      // Enable block support.
+      "-fblocks",
 
-          // Enable block support.
-          "-fblocks",
+      languageVersion.preprocessorDefinition("__swift__", {10000, 100, 1}),
 
-          languageVersion.preprocessorDefinition("__swift__", {10000, 100, 1}),
+      "-fretain-comments-from-system-headers",
 
-          "-fretain-comments-from-system-headers",
+      SHIMS_INCLUDE_FLAG, searchPathOpts.RuntimeResourcePath,
+  });
 
-          SHIMS_INCLUDE_FLAG, searchPathOpts.RuntimeResourcePath,
-      });
+  if (LangOpts.EnableObjCInterop) {
+    invocationArgStrs.insert(invocationArgStrs.end(),
+                             {"-x", "objective-c", "-std=gnu11", "-fobjc-arc"});
+    if (!triple.isOSDarwin())
+      invocationArgStrs.insert(invocationArgStrs.end(), {"-fobjc-runtime=ios"});
+  } else {
+    invocationArgStrs.insert(invocationArgStrs.end(), {"-x", "c", "-std=gnu11"});
+  }
+
+  // Get the version of this compiler and pass it to C/Objective-C declarations.
```

Comment:
This part does only belong in the Darwin section, because the compiler version numbers are only set in official Apple builds. I'd rather not have people start using that for other purposes at this point in time.

Reply:
cc @compnerd - this is part of your change

Reply:
@jrose-apple sure, thats easy enough to move.  However, that does seem useful to have in general.  Any reason to not always have the swift compiler version made available?

Reply:
We don't set this value at all for open source builds. (This is the "900.0.30" thing, not "4.0.2".)

Reply:
We barely even use it at Apple; there's a chance we might take it out altogether.

Reply:
@jrose-apple SGTM, I'll remove that